ggml-sycl.cpp

```diff
@@ -17031,9 +17263,13 @@ GGML_CALL static bool ggml_backend_sycl_supports_op(ggml_backend_t backend, cons
                     return false;
                 }
                 ggml_type a_type = a->type;
-                if (a_type == GGML_TYPE_IQ4_NL || a_type == GGML_TYPE_IQ2_S ||
-                    a_type == GGML_TYPE_IQ4_XS) {
-                    return false;
+
+                if (a_type == GGML_TYPE_IQ2_XXS || a_type == GGML_TYPE_IQ2_XS || a_type == GGML_TYPE_IQ3_XXS ||
+                    a_type == GGML_TYPE_IQ1_S   || a_type == GGML_TYPE_IQ4_NL || a_type == GGML_TYPE_IQ3_S   ||
+                    a_type == GGML_TYPE_IQ2_S   || a_type == GGML_TYPE_IQ4_XS) {
+                    if (b->ne[1] == 1 && ggml_nrows(b) > 1) {
+                        return false;
+                    }
```

Comment:
```suggestion
                // No support in mmvq or other methods
                if (a_type == GGML_TYPE_IQ1_M || a_type == GGML_TYPE_IQ4_NL || a_type == GGML_TYPE_IQ4_XS)
                    return false;

                // Support in mmvq
                if (a_type == GGML_TYPE_IQ2_XXS || a_type == GGML_TYPE_IQ2_XS || a_type == GGML_TYPE_IQ3_XXS ||
                    a_type == GGML_TYPE_IQ1_S   || a_type == GGML_TYPE_IQ3_S  || a_type == GGML_TYPE_IQ2_S   ) {
                    // condition for using mmvq
                    if (b->ne[1] > 1 || a->ne[0] % GGML_SYCL_DMMV_X != 0) {
                        return false;
                    }
```
The problem of asserts within non mmvq functions for the quantization types continues to exist. 

I recommend these changes and have found for them to allow the `test-backend-ops` to run correctly for intel and nvidia hardware. It essentially breaks it down to quantization types we do not support in any way. Types we support for mmvq and then the criteria to use mmvq.